sklearn/linear_model/_glm/glm.py

```diff
@@ -160,48 +184,14 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Fitted model.
         """
-        check_scalar(
-            self.alpha,
-            name="alpha",
-            target_type=numbers.Real,
-            min_val=0.0,
-            include_boundaries="left",
-        )
-        if not isinstance(self.fit_intercept, bool):
-            raise ValueError(
-                "The argument fit_intercept must be bool; got {0}".format(
-                    self.fit_intercept
-                )
-            )
+        self._validate_params()
+
         if self.solver not in ["lbfgs"]:
             raise ValueError(
                 f"{self.__class__.__name__} supports only solvers 'lbfgs'; "
                 f"got {self.solver}"
             )
         solver = self.solver
```

Comment:
I think we can remove this local variable and use `self.solver` directly when needed inside the `fit` method:

```suggestion
```